### PR TITLE
Add collection type filtering for manifestations (by-ccq)

### PR DESCRIPTION
**Pull request by abartov**
## Summary

Implements filtering of manifestations by their containment in volume or periodical_issue collections.

This PR addresses issue `by-ccq` and adds two new boolean fields to the ManifestationsIndex:
- `in_volume`: Set to true if manifestation is in a collection of type 'volume' (directly or through parent tree)
- `in_periodical`: Set to true if manifestation is in a collection of type 'periodical_issue' (directly or through parent tree)

### Model Layer
- Added `in_volume?` and `in_periodical?` methods to Manifestation model
- These methods traverse the collection parent tree to determine containment
- Properly handle nested collections and circular reference prevention

### Elasticsearch/Chewy
- Added `in_volume` and `in_periodical` boolean fields to ManifestationsIndex
- Migration automatically reindexes all manifestations on deployment

### Controller & Service
- Updated `build_es_filter_from_filters` in ManifestationController to handle collection type filters
- Implemented filter logic in SearchManifestations service
- Supports three filter options: "in a volume", "in a periodical", "uncollected"
- All three checked by default (shows all manifestations)
- If only "uncollected" is selected, filters for manifestations with both fields false

### UI
- Added checkbox group in /works browse view for collection type filtering
- I18n translations added for English and Hebrew

### Testing
- Comprehensive RSpec tests for `in_volume?` and `in_periodical?` methods
- Tests cover:
  - Direct containment in volume/periodical
  - Nested sub-collections
  - Deeply nested collections
  - Multiple collections
  - Edge cases (no collections, wrong collection types)

## Test Results

All tests pass:
- New model tests: 14 examples, 0 failures
- Full test suite: 0 failures

## Migration Notes

The migration will automatically run `ManifestationsIndex.reset!` to populate the new fields.
This may take a few minutes depending on the number of manifestations in the database.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/services/search_manifestations.rb [hunk 1]
```diff
@@ -42,6 +42,34 @@ def call(sort_by, sort_dir, filters)
...
+          # Uncollected: both in_volume and in_periodical are false
+          should_clauses << { bool: { must: [
+            { term: { in_volume: false } },
+            { term: { in_periodical: false } }
+          ] } }
+        end
+
+        filter << { bool: { should: should_clauses, minimum_should_match: 1 } }
+      end
+    end
```
[the remark this answers is not in the document]
Comment (abartov) [reply app/services/search_manifestations.rb]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/projectbenyehuda/bybe/pull/809#discussion_r2632539048)
